provider/provider.go

```diff
@@ -124,58 +123,84 @@ func (p *dockerNativeProvider) StreamInvoke(
 // the provider inputs are using for detecting and rendering diffs.
 func (p *dockerNativeProvider) Check(ctx context.Context, req *rpc.CheckRequest) (*rpc.CheckResponse, error) {
 	urn := resource.URN(req.GetUrn())
-	label := fmt.Sprintf("%s.Create(%s)", p.name, urn)
+	label := fmt.Sprintf("%s.Check(%s)", p.name, urn)
 	logging.V(9).Infof("%s executing", label)
 
 	inputs, err := plugin.UnmarshalProperties(req.GetNews(), plugin.MarshalOptions{
-		KeepUnknowns: false,
+		KeepUnknowns: true,
 		SkipNulls:    true,
 	})
 	if err != nil {
 		return nil, err
 	}
 
-	// Set defaults
+	buildOnPreview := marshalBuildOnPreview(inputs)
+	inputs["buildOnPreview"] = resource.NewBoolProperty(buildOnPreview)
+
 	build, err := marshalBuildAndApplyDefaults(inputs["build"])
 	if err != nil {
 		return nil, err
 	}
+	// Set the resource inputs to the default values
+	var knownDockerfile bool
+	if inputs["build"].IsNull() {
+		inputs["build"] = resource.NewObjectProperty(resource.PropertyMap{
+			"dockerfile": resource.NewStringProperty(build.Dockerfile),
+			"context":    resource.NewStringProperty(build.Context),
+		})
+		knownDockerfile = true
+	} else {
+		// We do not want to set these fields if their values are Unknown.
+		if !inputs["build"].ObjectValue()["dockerfile"].ContainsUnknowns() {
```

Comment:
It looks like if `inputs["build"]` is computed, then this will panic. Am I missing a check somewhere?

Reply:
I did think about this and was having a difficult time coming up with a practical example of a situation where a program would pass in the _entire build object_ as Unknown. 
It's not impossible, and definitely worth a safety check, but my thinking was, in most situations unknowns would be passed to the concrete fields, not the overall build object (and I haven't seen any bugs filed that showed this kind of pattern).

Reply:
It's weird but definitely possible! I've done things like this:

```typescript
build: thing.arn.apply((arn) => {
  return {
    dockerfile: "Dockerfile",
    args: {
      THE_ARN: arn
    }
  }
}) 
```
Sure you could have done that apply deeper in the object... but anything is possible :)

I added a small test to sanity check everything still works when build is computed and it caught a few panics... whew!